backend/remote-state/s3/backend_state.go

```diff
@@ -147,7 +157,17 @@ func (b *Backend) path(name string) string {
 		return b.keyName
 	}
 
-	return strings.Join([]string{keyEnvPrefix, name, b.keyName}, "/")
+	return b.keyName + keyEnvPrefix + name
```

Comment:
(possibly unrelated to this PR)
Probably a nitpick, but worth mentioning - the `key` attribute in the S3 `backend` block now has a different meaning, so it should probably be called `prefix` rather than `key`.

TL;DR this doesn't make sense anymore:

```hcl
backend "s3" {
    bucket  = "tf-test-bucket"
    key     = "something/terraform.tfstate"
    region  = "us-west-2"
  }
```